decimal_test.go

```diff
@@ -768,19 +768,74 @@ func TestJSONEncoding(t *testing.T) {
 			x := sign + test
 			var tx Decimal
 			tx.SetString(x)
-			b, err := json.Marshal(&tx)
+			b, err := tx.MarshalText()
 			if err != nil {
 				t.Errorf("marshaling of %s failed: %s", &tx, err)
 				continue
 			}
 			var rx Decimal
-			if err := json.Unmarshal(b, &rx); err != nil {
+			if err := rx.UnmarshalText(b); err != nil {
 				t.Errorf("unmarshaling of %s failed: %s", &tx, err)
 				continue
 			}
 			if rx.CmpTotal(&tx) != 0 {
-				t.Errorf("JSON encoding of %s failed: got %s want %s", &tx, &rx, &tx)
+				t.Errorf("text encoding of %s failed: got %s want %s", &tx, &rx, &tx)
 			}
 		}
 	}
 }
+
+func TestJSONUnmarshaling(t *testing.T) {
```

Comment:
Instead of these two tests, let's copy https://github.com/golang/go/blob/d0285161f0efb1837ba0942263c33e3d53f59ece/src/math/big/intmarsh_test.go#L100 (but with the list in TestJSONEncoding, which will have to be put at the top level) because it tests the text stuff and is a good pattern from the standard library.